core/src/main/scala/tech/beshu/ror/accesscontrol/factory/RawRorConfigBasedCoreFactory.scala

```diff
@@ -346,13 +352,17 @@ class RawRorConfigBasedCoreFactory(rorMode: RorMode)
             }
         }
       } yield {
-        implicit val loggingContext: LoggingContext = LoggingContext(obfuscatedHeaders)
         val blocks = blocksNel.map(_.block)
         blocks.toList.foreach { block => logger.info("ADDING BLOCK:\t" + block.show) }
         val localUsers: LocalUsers = {
-          val fromUserDefs = LocalUsers(users = Set.empty, unknownUsers = userDefs.items.map(_.usernames).nonEmpty)
+          val fromUserDefs = localUsersFromUserDefs(userDefs)
+          val fromImpersonatorDefs =
+            impersonationDefs.items
```

Comment:
let's move the logic to a separate, private method (like we did in case of `localUsersFromUserDefs`)